src/pages/index.js

```diff
@@ -73,57 +68,195 @@ function handleImageClick(cardData) {
 const cardSection = new Section(
   {
     items: initialCards,
-    renderer: renderCard,
+    renderer: (item) => {
+      const cardELement = createCard(item);
+      cardSection.addItem(cardELement);
+    },
   },
-  cardListEl
+  ".cards__list"
 );
 
-cardSection.renderItems();
+//Event Listeners
 
-// Render Cards
-function renderCard(item, method = "addItem") {
-  const cardElement = createCard(item);
-  cardSection[method](cardElement);
-}
+addNewCardButton.addEventListener("click", () => {
+  addCardFormPopup.open();
+});
+
+profileEditButton.addEventListener("click", () => {
+  // editProfileFormValidator.resetValidation();
+  const inputData = userInfo.getUserInfo();
+  // console.log(inputData);
+  profileNameInput.value = inputData.name;
+  profileDescriptionInput.value = inputData.about;
+  profileEditPopup.open();
+});
+
+addCardFormValidator.enableValidation();
+editProfileFormValidator.enableValidation();
 
 //Functions
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
+  return card.generateCard();
 }
 
 function handleProfileEditSubmit(inputData) {
-  userInfo.setUserInfo({
-    name: inputData.name,
-    job: inputData.description,
-  });
+  profileEditPopup.renderLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```